src/routes/business/business-address-check-routes.js

```diff
@@ -1,46 +1,31 @@
+import { businessAddressCheckPresenter } from '../../presenters/business/business-address-check-presenter.js'
+import { flashNotification } from '../../utils/notifications/flash-notification.js'
+import { setSessionData } from '../../utils/session/set-session-data.js'
+
 const getBusinessAddressCheck = {
   method: 'GET',
   path: '/business-address-check',
   handler: (request, h) => {
-    const address1 = request.state.address1 || ''
-    const address2 = request.state.address2 || ''
-    const city = request.state.city || ''
-    const county = request.state.county || ''
-    const postcode = request.state.postcode || ''
-    const country = request.state.country || ''
-
-    return h.view('business/business-address-check', {
-      address1,
-      address2,
-      city,
-      county,
-      postcode,
-      country
-    })
+    const businessDetailsData = request.yar.get('businessDetails')
+
+    const sessionData = request.yar.get('businessAddress')
+
+    const pageData = businessAddressCheckPresenter(businessDetailsData, sessionData)
+
+    return h.view('business/business-address-check', pageData)
   }
 }
 
 const postBusinessAddressCheck = {
   method: 'POST',
   path: '/business-address-check',
   handler: (request, h) => {
-    const address1 = request.state.address1 || ''
-    const address2 = request.state.address2 || ''
-    const city = request.state.city || ''
-    const county = request.state.county || ''
-    const postcode = request.state.postcode || ''
-    const country = request.state.country || ''
+    const sessionData = request.yar.get('businessAddress')
+    setSessionData(request.yar, 'businessDetails', 'businessAddress', sessionData)
+
+    flashNotification(request.yar, 'Success', 'You have updated your business address')
```

Comment:
Expected pattern is something like
await updateBusinessAddressChangeService(request.yar)